stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -296,3 +388,66 @@ def is_greater_eq(value, other):
                     # compute how much packaging we can get
                     take = (need // pack_quantity) * pack_quantity
                     need = yield location, location_quantity, take, None, None
+
+    @api.model
+    def _get_requested_comparisons(self, rounding):
+        value = self.tolerance_requested_value
+        if value == 0.0:
+            return lambda product_qty, need: product_qty == need
+        computation = self.tolerance_requested_computation
+        if self.tolerance_requested_limit == "upper_limit":
+
+            def wrap(product_qty, need):
+                if computation == "percentage":
+                    return need + rounding < product_qty <= need * (100 + value) / 100
+                # computation == "absolute"
+                else:
+                    return need + rounding < product_qty <= need + value
```

Comment:
I don't understand the comparison formulas.
You say in readme `select a lot with higher qty (Tolerance on = Upper Limit")`. I understand that in this case you want a quant that has more stock than the need, i.e. `need <= lot_qty`. You need to use float_compare and provide the rounding.

Reply:
Or I wrongly understood what you try to achieve. Looking at your formula, you try in fact to reserve a full lot. Like we have full packaging and full bin where the quantity matches the need, or where the quantity is a bit more (upper) or less (lower) than the need.
If this is the case, then the rule should be definitely renamed to `full_lot` and the readme clarified.
Still the formulas must be fixed.
* No tolerance: `float_compare(need, quantity, rounding) == 0`
* Upper Limit: lot with higher qty than required: `float_compare(need, quantity, rounding) <= 0 and float_compare(quantity - need, tolerance, rounding) <= 0`

Reply:
This is not resolved